tests/testSchema.py

```diff
@@ -106,6 +106,52 @@ def testKey(name, key):
         keys2 = map(lambda x: x.key, schema4.asList())
         self.assertEqual(keys, keys2)
 
+    def testUnits(self):
+        schema = lsst.afw.table.Schema();
+	# first insert some valid units
+        try:
+            schema.addField("a", type="I", units="pixel")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.addField("b", type="I", units="m2")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.addField("c", type="I", units="electron / adu")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.addField("d", type="I", units="kg m s^(-2)")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.addField("e", type="I", units="GHz / Mpc")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.addField("f", type="Angle", units="deg")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.addField("g", type="Angle", units="rad")
+        except ValueError:
+            self.fail("addField raised ValueError unexpectedly!")
+        try:
+            schema.checkUnits()
+        except ValueError:
+            self.fail("checkUnits raised ValueError unexpectedly!")
```

Comment:
I don't think all the try/except blocks here are actually helpful - if we just let the exception propagate up, the test will still fail, and the traceback in that case might give whoever is debugging the failure a bit more useful information.